Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.EventSystems;
+using System.Collections.Generic;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Core.EventDatum.Input;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Devices;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Core.Managers;
+using Microsoft.MixedReality.Toolkit.SDK.Input;
+using Microsoft.MixedReality.Toolkit.SDK.Input.Handlers;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Receivers
+{
+    /// <summary>
+    /// An interaction receiver is simply a component that attached to a list of interactable objects and does something
+    /// based on events from those interactable objects.  This is the base abstract class to extend from.
+    /// </summary>
+    public abstract class InteractionReceiver : BaseInputHandler,
+        IMixedRealityFocusChangedHandler,
+        IMixedRealityGestureHandler,
+        IMixedRealityInputHandler,
+        IMixedRealityGestureHandler<Vector2>,
+        IMixedRealityGestureHandler<Vector3>,
+        IMixedRealityGestureHandler<Quaternion>
+
+    {
+        #region Public Members
+        /// <summary>
+        /// List of linked interactable objects to receive events for
+        /// </summary>
+        [SerializeField]
+        [ Tooltip("Target interactable Object to receive events for")]
+        private List<GameObject> interactables = new List<GameObject>();
+
+        public List<GameObject> Interactables
+        {
+            get { return interactables; }
+            set { value = interactables; }
```

Comment:
I'm not sure we want to allow people to set a new list. We can make this a getter only as it doesn't prevent people from modifying the list.

Reply:
@StephenHodgson Setting this list of Interactables from the inspector by dragging & dropping objects from the scene is the crucial elements of the Receiver. Same for the 'Targets' 
You can find the example in the 'InteractableObjectExample' scene.

![2018-09-18 22_13_27-settings](https://user-images.githubusercontent.com/13754172/45732204-19f56f00-bb90-11e8-8d95-4fa2fcac106a.png)


![2018-09-18 22_12_27-settings](https://user-images.githubusercontent.com/13754172/45732184-06e29f00-bb90-11e8-9be2-0adff4509fcf.png)

Reply:
You should still be able to add/remove items from the list as well. Having a get only accessor just means someone can't make a new list and set it.

Reply:
Right, using List.Add() / Remove(). Yes, I can't think about an example of replacing the list with a new one.

Reply:
This is really good to know. I always thought any modification of a list required the setter as well. I’d be happy to change.